File: LeetCode/1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_dict = {}
        for i, v in enumerate(nums):
            comp_num = target - v
            if comp_num in num_dict:
                return [i, num_dict[comp_num]]
            num_dict[v] = i
            
        return []
            

File: LeetCode/1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const myMap = new Map();

    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        const comp = target - num;

        if (myMap.has(comp)) {
            return [myMap.get(comp), i];
        };
        
        myMap.set(num, i);
    }
};